### ISSUE TITLE
[ProgressMonitors] Dispose is not called on the right sync context

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#4751]
Dispose should be called on the right synchronization context, but it is not possible to do so at the moment.

https://github.com/mono/monodevelop/blob/bbc80493465ff7c7e32b51c3d4b25ed751859610/main/src/core/MonoDevelop.Core/MonoDevelop.Core/ProgressMonitor.cs#L134

Dispose context is controlled by the caller, so we can't marshal to any other sync context.

Consider breaking API and changing to:
```
public void Dispose ()
{
...
}

protected virtual void OnDisposed()
{
...
}
```

And using the synchronization context. This is currently a bad case for UI progress monitors disposed from bg threads.

> VS bug [#612543](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/612543)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.690 (master)

Author: therzok
Commit: fdaef70c331ffe5580ce9d88b9aba4a3aaa1f348 (mono/monodevelop)

**Comment by Therzok** [comment]
Shouldn't have been closed.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-18] Comment by **xamarinc**



Fixed in version 8.0.0.778 (release-8.0)



Author: Marius Ungureanu  
 Commit: 276c2bcdc505d249d48f953d6233a3b7a1f1229e (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-20] Comment by **xamarinc**



Fixed in version 8.0.0.1001 (master)



Author: Marius Ungureanu  
 Commit: 276c2bcdc505d249d48f953d6233a3b7a1f1229e (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-21] Comment by **xamarinc**



Fixed in version 8.0.0.1738 (release-8.0)



Author: Marius Ungureanu  
 Commit: 276c2bcdc505d249d48f953d6233a3b7a1f1229e (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-25] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: Marius Ungureanu  
 Commit: 276c2bcdc505d249d48f953d6233a3b7a1f1229e (xamarin/monodevelop)
